### PR TITLE
High contrast: Multiple component support for mouse over states

### LINKED ISSUE #4533: Components need to support mouse hover in high contrast
These controls currently do not respond to mouse hover in high contrast themes:

1. Button - Needs to change the border to system color "highlight"
2. Breadcrumb - See above. I believe this will inherit from Button
3. Checkbox - See Button. Also technically it would be better if the background and checkmark reversed colors on selection.
4. Coachmark has all kinds of problems in HC
5. ChoiceGroup - See Button
6. Link - underline on hover would be great
7. Pivot - Should show underline on hover
8. Rating - Stars should change to system color "highlight" on hover
9. Searchbox - See button
10. Spin button - Remove horrific dashed line. How did that get through?
image
11. Toggle - See Button

### PR DESCRIPTION
**Pull request by lynamemi**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #4533  
- [X] Include a change request file using `$ npm run change`

#### Description of changes

I grouped some of the simpler component states from the issue:

**Breadcrumb**:
breadcrumb-hover-hc

**Button** (made this general change across many buttons):
button-hover-hc

**Link**:
link-hover-hc

**Pivot**:
pivot-hover-hc

**SearchBox**:
searchbox-hover-hc

**SpinButton**:
spinbutton-hover-hc

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by cliffkoh** [comment]
Just adding a note that command bar has been promoted from experiments to component in 6.0 (presently in the `new_6.0` branch). Command bar specific changes will need to be manually applied in that branch to not be "lost" with 6.0.

**Comment by lynamemi** [comment]
@cliffkoh Is that something you want me to do or are you doing a sweep of those updates?

**Comment by cliffkoh** [comment]
Could you split the command bar changes out from this PR and create a separate PR for the command bar in new_6.0? Fortunately, the change looks simple to port. Amongst other things (such as 6.0 specific features/changes do not keep getting delayed further and further back due to endless port-work), this will allow the commit moving forward to be correctly attributed to you :)

File: packages/office-ui-fabric-react/src/components/Link/Link.styles.ts [hunk 1]
```diff
@@ -58,7 +59,12 @@ export const getStyles = (props: ILinkStyleProps): ILinkStyles => {
       !isDisabled && {
         selectors: {
           '&:active, &:hover, &:active:hover': {
-            color: semanticColors.linkHovered
+            color: semanticColors.linkHovered,
+            selectors: {
+              [HighContrastSelector]: {
+                textDecoration: 'underline'
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/Link/Link.styles.ts]: Can you change this to border-bottom instead?

The reason is, I think for places where we're adding borders, we should just change it to be transparent borders and remove the high contrast selector. That way, it doesn't interfere with the default UX, while at the same time it'll actually be visible in high contrast mode in FireFox.

@betrue-final-final thoughts?

File: packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts [hunk 2]
```diff
@@ -195,7 +195,8 @@ export const getStyles = memoizeFunction((
       outline: '2px dashed transparent',
       selectors: {
         [HighContrastSelector]: {
-          borderColor: 'Highlight'
+          borderColor: 'Highlight',
+          outline: 'none'
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts]: why not keep using outline here?

Comment (lynamemi) [reply packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts]: The outline was creating a lot of noise, and Ben wanted it gone. See this screenshot:
image

**Comment by cliffkoh** [comment]
Thank you 👏

File: packages/office-ui-fabric-react/src/components/Pivot/Pivot.scss [hunk 3]
```diff
@@ -40,6 +40,13 @@
 
   &:hover {
     cursor: pointer;
+
+    @include high-contrast {
+      &::before {
+        box-sizing: border-box;
+        border-bottom: 2px solid Highlight;
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/Pivot/Pivot.scss]: set this to be a permanent hover style outside of @include high-contrast, and then change the color for just IE

**Pull request merged**